Fix #581 - Set map bounds using screen size in zoomToRegion()

therajanmaurya opened the pull request:
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Sign the [electronic OBA ICLA](https://docs.google.com/forms/d/12jV-ByyN186MuPotMvxJtNKtSaGGTnEHm8rXomM2bm4/viewform)

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `gradlew connectedObaGoogleDebugAndroidTest connectedObaAmazonDebugAndroidTest` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

- Add : Dynamic Map size, according to device screen size .

therajanmaurya commented:
Hi @barbeau Please review.

And I did not sign the ICLA till now. Is it necessary for this PR to merge to sign the ICLA.

barbeau commented:
ICLA manually verified :+1:

barbeau commented on onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java:
```diff
@@ -654,8 +654,13 @@ public void zoomToRegion() {
+            //Setting the width and height of your screen
+            int width = getResources().getDisplayMetrics().widthPixels;
+            int height = getResources().getDisplayMetrics().heightPixels;
+            int padding = (int) (width * 0.12); // offset from edges of the map 12% of screen
```
Any reason in particular that exactly 12% is used here?

barbeau commented:
@therajanmaurya Looks like you modified the file in the path `/src/amazon/java/` - you'll need to make these changes to file in `/src/google/java/` instead, and then run the app - the code for `/src/amazon/java/` (for Amazon FirePhone) will then be automatically generated.  The end result should be edits to `BaseMapFragment` in both paths.

therajanmaurya commented:
@barbeau it's my mistake, Sorry for that to forget to change the build variant.
Now it's done and I updated the PR with build variant obaGoogleDebug.
I have changed the padding = 0 and documented the code, what's the use of padding = 0.
Reference for more about the padding = 0 : http://stackoverflow.com/a/27937256/3176277

barbeau commented:
@therajanmaurya Are you actually installing this patch on an emulator or device?  The build script should generate the edits to `BaseMapFragment` for the Amazon build variant as well.  So you have have two files edited in this PR is all goes well.

therajanmaurya commented:
@barbeau Yes, I am installing this patch to Moto G 3 turbo Edition. 
and I have checked to change the build variant to Amazon and yes, It's showing BaseMapFragment is changed as change occur in Google variant.

barbeau commented:
You should just leave the build variant on  obaGoogleDebug.  Sounds like
the Amazon changes didn't get committed with the Google changes.  Can you
please update PR with single commit that includes changes to both files (as
generated by build script)?

On Jul 18, 2016 1:48 PM, "Rajan Maurya" <notifications@github.com> wrote:

> @barbeau <https://github.com/barbeau> Yes, I am installing this patch to
> Moto G 3 turbo Edition.
> and I have checked to change the build variant to Amazon and yes, It's
> showing BaseMapFragment is changed as change occur in Google variant.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OneBusAway/onebusaway-android/pull/598#issuecomment-233404106>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA4pLRXBr17YoFuhjK2qevgTWJdYt6mZks5qW7xzgaJpZM4JOa82>
> .
>

therajanmaurya commented:
you mean, make the commit again after changing the build variant to Amazon and then squash the both commit.

barbeau commented:
No, don't change to Amazon variant.  Leaving it on obaGoogleDebug.  Build
the app, and then squash any changes from master into one commit for this
PR.

The only reason for changing to the Amazon variant is if you have a Fire
Phone and want to run the app on it. If you are building on a normal Google
device, don't change to the Amazon variant.

On Jul 18, 2016 2:07 PM, "Rajan Maurya" <notifications@github.com> wrote:

> you mean, make the commit again after changing the build variant to Amazon
> and then squash the both commit.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OneBusAway/onebusaway-android/pull/598#issuecomment-233409659>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA4pLbdOPw8hNLkWYV2ASYar58xcN0gkks5qW8DegaJpZM4JOa82>
> .
>

therajanmaurya commented:
@barbeau it's done as you said. commit the changes of obaGoogleDebug and did not change to Amazon variant and with Squash !

barbeau commented:
@therajanmaurya Somethings not going right - I'm still only seeing one file changed in this commit.  There should be two, one for each build variant.  You get this by following these steps:

1. Checkout master branch
1. Checkout new branch for this PR
1. Select `obaGoogleDebug` build variant
1. Make changes to the `BaseMapFragment` class in the `/google/` file path
1. Build and run app

At this point, in Android Studio you should see changes to `BaseMapFragment` in both the google and amazon file paths.

1. Commit these changes and force push to your branch for this PR.

therajanmaurya commented:
Thanks @barbeau for the guideline. It's done now and perfects.
Actually, I was doing git commit before running on the device because When I was done first then I run on the device and on the second time not running on the device and making git commit.

barbeau commented:
@therajanmaurya Thanks, looks better know.  I'll need to test, and it might be at least a few days until I get to that.

therajanmaurya commented:
@barbeau It's Ok, Can you tell me more about the build script ?

barbeau commented:
Sure.  For discussion off-topic from this PR/issue I'd head over to Slack,
and we can discuss in #android room there.

Sean J. Barbeau, Ph.D.
sjbarbeau@gmail.com
http://www.linkedin.com/in/seanbarbeau

On Mon, Jul 18, 2016 at 3:58 PM, Rajan Maurya <notifications@github.com>
wrote:

> @barbeau <https://github.com/barbeau> It's Ok, Can you tell me more about
> the build script ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OneBusAway/onebusaway-android/pull/598#issuecomment-233440392>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA4pLYGi8vxqM7kUf55QngysfNE_HsG3ks5qW9rsgaJpZM4JOa82>
> .
>

barbeau commented:
@cagryInside Could you please review/test this PR and see if all looks ok?

barbeau commented:
Actually, looking at the crash reports, we're seeing a large number of reports for this issue (115 in the last week).  @cagryInside Could you review today?  I'd like to get a hotfix out for this.

cagryInside commented:
> Actually, looking at the crash reports, we're seeing a large number of reports for this issue (115 in the last week). @cagryInside Could you review today? I'd like to get a hotfix out for this.
>  Show all checks

@barbeau Sure, I am looking at it now.

cagryInside commented:
@barbeau I couldn't reproduce the issue. However, I tested this PR with different phones. All are working fine. Further, based on the stackoverflow entry that you posted (http://stackoverflow.com/a/30497039/937715) and my internet research on this issue, this is the accepted fix.

barbeau commented:
Thanks @cagryInside for reviewing!  Looks good to me too.  Thanks @therajanmaurya for the contribution!
